UI/Pages/Shortlist.cshtml.cs

```diff
@@ -0,0 +1,172 @@
+
+        if (!Data.Validation.IsValid)
+            foreach (var error in Data.Validation.Errors)
+                ModelState.AddModelError($"Data.{error.PropertyName}", error.ErrorMessage);
+    }
+
+    public record Query : SearchModel, IRequest<ResultsModel>;
+
+    public record ResultsModel : SearchModel
+    {
+        public ResultsModel() { }
+        public ResultsModel(SearchModel query) : base(query) { }
+
+        public TuitionPartnersResult? Results { get; set; }
+        public FluentValidationResult Validation { get; internal set; } = new();
+    }
+
+    private class Validator : AbstractValidator<Query>
+    {
+        public Validator()
+        {
+            RuleFor(m => m.Postcode)
+                .Matches(StringConstants.PostcodeRegExp)
+                .WithMessage("Enter a valid postcode")
+                .When(m => !string.IsNullOrEmpty(m.Postcode));
+        }
+    }
+
+    public class Handler : IRequestHandler<Query, ResultsModel>
+    {
+        private readonly ILocationFilterService _locationService;
+        private readonly ITuitionPartnerService _tuitionPartnerService;
+        private readonly INtpDbContext _db;
+        private readonly ILogger<TuitionPartner> _logger;
+
+        public Handler(ILocationFilterService locationService, ITuitionPartnerService tuitionPartnerService, INtpDbContext db, ILogger<TuitionPartner> logger)
+        {
+            _locationService = locationService;
+            _tuitionPartnerService = tuitionPartnerService;
+            _db = db;
+            _logger = logger;
+        }
+
+        public async Task<ResultsModel> Handle(Query request, CancellationToken cancellationToken)
+        {
+            var queryResponse = new ResultsModel(request);
+
+            var searchResults = await GetSearchResults(GetShortlist(), request, cancellationToken);
+
+            return searchResults switch
+            {
+                SuccessResult => queryResponse with
+                {
+                    Results = searchResults.Data,
+                },
+                Domain.ValidationResult error => queryResponse with
+                {
+                    Validation = new FluentValidationResult(error.Failures)
+                },
+                ErrorResult error => queryResponse with
+                {
+                    Validation = new FluentValidationResult(new[]
+                    {
+                        new ValidationFailure("Postcode", error.ToString()),
+                    }),
+                },
+                _ => queryResponse with { Validation = UnknownError() },
+            };
+
+            static FluentValidationResult UnknownError() =>
+                new(new[] { new ValidationFailure("", "An unknown problem occurred") });
+        }
+
+        private int[] GetShortlist()
+        {
+            //TODO - integrate with shortlist selection
+            var tuitionPartnersIds = new int[] { 124, 125, 126, 127, 129, 131 };
```

Comment:
Using the row ids will mean that shortlists will become invalid after a data import (as the ids all change). I wonder is we should use the seoUrl for the time being as that's the key we use in most other places?

Reply:
Quite possibly, sounds like a good shout - I have just mocked this up for now, but we can switch it to seoUrl when Kola passes me the info over via cookie.

Reply:
Updated to use SeoURL.